sprockets_postgres.py

```diff
@@ -361,18 +361,27 @@ async def postgres_connector(self,
                         as cursor:
                     yield PostgresConnector(
                         cursor, on_error, on_duration, timeout)
-        except (asyncio.TimeoutError, psycopg2.OperationalError) as err:
+        except (asyncio.TimeoutError,
+                psycopg2.OperationalError,
+                RuntimeError) as err:
```

Comment:
Might make sense to change the raised `RuntimeError`s in sprockets_postgres to a custom exception. Don't care what it is or if it's documented, just would like to avoid a situation where the app using sprockets_postgres raises a `RuntimeError` within the context of `postgres_connector`. 

Ex.
```
async with postgres_connector(...):
    raise RuntimeError('not related to database stuff
```

Would be handled by the change in this PR

Reply:
I believe that the flow is to take that exception and wrap it in a `ConnectionException` and it is caught because that's what aiopg.pool.Pool is raising when the pool is closed.

Reply:
Clearer example of what I'm talking about:

This handler should return an HTTP 418, but instead fails with an HTTP 503. My `RuntimeError('andrew debugging')` is erroneously transformed into a `sprockets_postgres.ConnectionException` and my handler fails to work as I intended it to. 
```
class AndrewHandler(web.RequestHandler):

    async def get(self):
        try:
            async with self.application.postgres_connector():
                raise RuntimeError('andrew debugging')
        except sprockets_postgres.ConnectionException:
            raise web.HTTPError(503)
        except RuntimeError:
            raise web.HTTPError(418)


class AndrewTestCase(TestCase):

    def test_it(self):
        response = self.fetch('/andrew')
        self.assertEqual(response.code, 418)
```

---

Is there a particular `RuntimeError` exception that prompted you to add handling here? sprockets_postgres raises it in a few spots, but I only see three instances of it in [aiopg](https://github.com/aio-libs/aiopg/blob/08fd8e3fecd69d300741bc954aaa96585efb5141/aiopg/pool.py). Not sure if the aiopg ones are relevant.

Reply:
The catching of RuntimeError in this method is around trying to acquire a connection from the pool and seeing `aiopg.pool.Pool` raise it here: https://github.com/aio-libs/aiopg/blob/08fd8e3fecd69d300741bc954aaa96585efb5141/aiopg/pool.py#L306

This is specifically at the connection acquisition phase and shouldn't cascade up to the request handler imo